zio-http-gen/src/main/scala/zio/http/gen/openapi/Config.scala

```diff
@@ -19,20 +23,55 @@ final case class Config(
    *
    * TODO: in the future we can consider an enum instead of boolean for different aliased types.
    *       e.g: scala 3 opaque types, neotype, prelude's newtype, etc'…
+   */,
+  fieldsNormalizationConf: NormalizeFields, /*
+   * OpenAPI can declare fields that have unconventional "casing" in scala,
+   * like snake_case, or kebab-case.
+   * This configuration allows to normalize these fields.
+   * The original casing will be preserved via a @fieldName("<original-name>") annotation.
    */
 )
 object Config {
 
+  final case class NormalizeFields(
+    enabled: Boolean /*
+     * If enabled, the generator will attempt to normalize field names to camelCase,
+     * unless original field is defined in the specialReplacements map.
+     */,
+    specialReplacements: Map[String, String], /*
+     * When normalization is enabled, a heuristic parser will attempt to normalize field names.
```

Comment:
Map can be used also when auto normalization is not enabled. I should amend the documentation to be clearer

Reply:
If `specialReplacements` can be used out side of normalization, does it belong in the normalization config?

Reply:
I meant of course it's part of normalization. but it overrides **auto** normalization (i.e: using the parser to construct a different term).
Normalization has 2 modes:
- manual replacements (user control exactly what to replace with what)
- automatic (a heuristic parser will try to do the best it can)

both fall under normalization.
Perhaps I should rename as: `automaticEnabled` & `manualReplacements`?

Reply:
Would turning the automated off lead to invalid code? Should it be always on? And the manual just be tried first, if the map is not empty?